pkg/audit/disqualification_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite"
+)
+
+// TestDisqualificationTooManyFailedAudits does the following:
+// * Create a failed audit report for a storagenode
+// * Record the audit report several times and check that the node isn't
+//	 disqualified until the audit reputation reaches the cut-off value.
+func TestDisqualificationTooManyFailedAudits(t *testing.T) {
+	var (
+		auditDQCutOff float64 = 0.4
+		alpha0        float64 = 1
+		beta0         float64 = 0
+	)
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, Reconfigure: testplanet.Reconfigure{
+			Satellite: func(log *zap.Logger, index int, config *satellite.Config) {
+				config.Overlay.Node.AuditReputationAlpha0 = alpha0
+				config.Overlay.Node.AuditReputationBeta0 = beta0
+				config.Overlay.Node.AuditReputationLambda = 1
+				config.Overlay.Node.AuditReputationWeight = 1
+				config.Overlay.Node.AuditReputationDQ = auditDQCutOff
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		err := planet.Satellites[0].Audit.Service.Close()
+		require.NoError(t, err)
+
+		var (
+			sat    = planet.Satellites[0]
+			nodeID = planet.StorageNodes[0].ID()
+			report = &audit.Report{
+				Fails: storj.NodeIDList{nodeID},
+			}
+		)
+
+		dossier, err := sat.Overlay.Service.Get(ctx, nodeID)
+		require.NoError(t, err)
+
+		require.Equal(t, alpha0, dossier.Reputation.AuditReputationAlpha)
+		require.Equal(t, beta0, dossier.Reputation.AuditReputationBeta)
+
+		prevReputation := calcReputation(dossier)
+
+		// Report the audit failure until the node gets disqualified due to many
+		// failed audits
+		iterations := 1
```

Comment:
I think this should start at 0. After the loop, you check if `iterations > 1`, which will happen after a single iteration

Reply:
No, `iteration++` is executed after the loop body finishes (without breaking/returning)

Reply:
Oh, did not realize